### ISSUE TITLE
Avoiding Amazon Highlight Limits

### ISSUE THREAD

**Issue by wclausen** [z1lc/zdone#181]
Amazon/publishers often impose limits on how much of a book can be highlighted. Note, the limitation is on what will show up in the notebook section, not what will be highlighted in the actual book text on your device, aka it affects export to readwise but won't affect highlighting in reading experience. The limit is usually around 10-15% of the text.

The problem here is that the highest quality books are likely to the be the books you highlight the most, thus making those highlights at risk of being lost/clipped. This has already happened for me with the "What Every Developer Should Know About HTTP" book. I'd like to establish a way around this.

Here's an idea I had that I think could work: When you fear you are getting close to the limit for highlighted content in a book, we have a little chrome extension that you can use to delete all highlights for the book from Amazon's web page that shows the highlights. This will free up space for future highlights. And since we have exported the highlights already, the old highlights will continue to be used during card generation.

Thoughts @z1lc?

**Comment by z1lc** [comment]
Yeah this sounds good to me. Something that would likely best live in the new https://github.com/z1lc/zdone-webextension. If we could figure out what exact % it stops at, then we could also safely delete highlights -- eg let's say that the level at which it stops is 10%, and by the time you use the extension you've highlighted 15%, then ideally the extension would only delete the first 10% of highlights but keep the remaining un-exported 5% as-is.

If we couldn't figure out the exact percentage, we could just take a look at what we have in the DB. aka send highlights that the extension identifies to some endpoint on zdone.co and see if it's already in the db or not, if so then delete, if not leave. In fact, maybe we cut out that process entirely -- why not just send highlights to zdone directly as part of the extension? would just be a question of deduping what readwise has, which shouldn't be too hard.
